homeassistant/components/yolink/climate.py

```diff
@@ -0,0 +1,115 @@
+    PRESET_NONE,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_COORDINATORS, ATTR_DEVICE_THERMOSTAT, DOMAIN
+from .coordinator import YoLinkCoordinator
+from .entity import YoLinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up YoLink Thermostat from a config entry."""
+    device_coordinators = hass.data[DOMAIN][config_entry.entry_id][ATTR_COORDINATORS]
+    entities = [
+        YoLinkClimateEntity(config_entry, device_coordinator)
+        for device_coordinator in device_coordinators.values()
+        if device_coordinator.device.device_type == ATTR_DEVICE_THERMOSTAT
+    ]
+    async_add_entities(entities)
+
+
+class YoLinkClimateEntity(YoLinkEntity, ClimateEntity):
+    """YoLink Climate Entity."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: YoLinkCoordinator,
+    ) -> None:
+        """Init YoLink Thermostat."""
+        super().__init__(config_entry, coordinator)
+        self._attr_unique_id = f"{coordinator.device.device_id}_climate"
+        self._attr_name = f"{coordinator.device.device_name} (Thermostat)"
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_fan_modes = [FAN_ON, FAN_AUTO]
+        self._attr_min_temp = -10
+        self._attr_max_temp = 50
+        self._attr_hvac_modes = [
+            HVACMode.COOL,
+            HVACMode.HEAT,
+            HVACMode.AUTO,
+            HVACMode.OFF,
+        ]
+        self._attr_preset_modes = [PRESET_NONE, PRESET_ECO]
+        self._attr_supported_features = (
+            ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        )
+
+    @callback
+    def update_entity_state(self, state: dict[str, Any]) -> None:
+        """Update HA Entity State."""
+        normal_state = state["state"]
+        self._attr_current_temperature = normal_state[ATTR_TEMPERATURE]
+        self._attr_current_humidity = normal_state["humidity"]
+        self._attr_target_temperature_low = normal_state["lowTemp"]
+        self._attr_target_temperature_high = normal_state["highTemp"]
+        self._attr_fan_mode = normal_state["fan"]
+        self._attr_hvac_mode = normal_state["mode"]
+        self._attr_hvac_action = normal_state["running"]
+        self._attr_preset_mode = (
+            PRESET_NONE if state["eco"]["mode"] == "on" else PRESET_ECO
+        )
+        self.async_write_ha_state()
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set HVAC mode."""
+        await self.call_device_api("setState", {"mode": hvac_mode.value})
```

Comment:
We need an inverted map here to map from Home Assistant mode to yolink device mode.